lib/mutant/mutator/node/define.rb

```diff
@@ -13,11 +13,40 @@ class Define < self
         # @api private
         def dispatch
           emit_arguments_mutations
+          emit_body_assignment_mutations
           emit_body(N_RAISE)
           emit_body(nil)
           emit_body_mutations if body
         end
 
+        # Emit mutations with optional arguments as assignments in method
+        #
+        # @return [undefined]
+        #
+        # @api private
+        def emit_body_assignment_mutations
+          arguments.children.each do |argument|
+            next unless n_optarg?(argument) && AST::Meta::Optarg.new(argument).used?
+
+            emit_body_prepend(s(:lvasgn, *argument))
+          end
+        end
+
+        # Emit valid body ASTs depending on instance body
+        #
+        # @param node [Parser::AST::Node]
+        #
+        # @return [undefined]
+        #
+        # @api private
+        def emit_body_prepend(node)
+          if body
+            emit_body(s(:begin, node, body))
+          else
+            emit_body(node)
+          end
+        end
```

Comment:
Lets push it down to private `Mutator::Node#emit_body_concat` ? And call it from here and the rescue mutator?